blac-siren/Maintenance-Tracker
# # 158145335 Build out API endpoints for resource CRUD

**What does this PR do?**
Refactor the code to use blueprints and sets up connection of the app to the database.
Show all the CRUD enpoints.

**Description of Task to be completed?**
### Api endpoints
| url | Method|  Description 
| --- | --- | --- 
| /api-v1/auth/register | POST | Registers new user 
| /api-v1/auth/login | POST | Handles POST request for /login
| /api-v1/users/requests | GET | Get every requests for logged in user
| /api-v1/users/requests/{id} | GET | Gets one of the request. 
| /api-v1/users/requests | POST | Create a new request
| /api-v1/users/requests/{requestId}  | PUT | Update an existing request
| /api-v1/users/requests/{requestId} | DELETE | Delete request by id

**How should this be manually tested?**
clone the repository
open the folder in your text editor or IDE
`git checkout -b new-branch`
`git fetch ft-Auth-create-request-158145335`

`source venv/bin/activate to run environmen`t
`pip install -r requirements.txt`
Database
Prerequisites: Should have Postgres installed
Open terminal and run
   ` $ CREATE DATABASE trackerapp;`
  `  $ \q   `
 `   $   table_db.py`
 `   $ python run.py`

**Any background context you want to provide?**
None

What are the relevant pivotal tracker stories?
158145335
